### PR TITLE
Fix a field name bug in cluster alert banners

### PR DESCRIPTION
**Pull request by bl-nero**
This bug prevented links from being properly rendered.

### CODE REVIEW

**Review by zmb3 — APPROVED** [review]

File: web/packages/teleport/src/Main/Main.tsx [hunk 1]
```diff
@@ -188,7 +188,7 @@ export function Main(props: MainProps) {
   const banners: BannerType[] = alerts.map(alert => ({
     message: alert.spec.message,
     severity: mapSeverity(alert.spec.severity),
-    link: alert.metadata.labels[LINK_DESTINATION_LABEL],
+    linkDestination: alert.metadata.labels[LINK_DESTINATION_LABEL],
```
Comment (zmb3) [inline web/packages/teleport/src/Main/Main.tsx:191]: Could typescript have helped us catch this error?

File: web/packages/teleport/src/Main/Main.tsx [hunk 2]
```diff
@@ -188,7 +188,7 @@ export function Main(props: MainProps) {
   const banners: BannerType[] = alerts.map(alert => ({
```
Comment (flyinghermit) [inline web/packages/teleport/src/Main/Main.tsx:188]:
```suggestion
  const banners: BannerType[] = alerts.map((alert): BannerType => ({
```

You should also add a return type for the callback function, `BannerType`  in this case.  That would have helped typescript to catch this issue earlier and prevent similar one in the future.

File: web/packages/teleport/src/Main/Main.tsx (hunk 1 above)
Comment (flyinghermit) [reply web/packages/teleport/src/Main/Main.tsx:191]: Typescript would have caught this but only if the callback function in the `alerts.map` had a return type defined as suggested in https://github.com/gravitational/teleport/pull/46840#discussion_r1769675377 below.

This is one of the weakness of typescripts' structural type system.

Comment (bl-nero) [reply web/packages/teleport/src/Main/Main.tsx:191]: Thanks!

**Review by flyinghermit — APPROVED** [review]

**Pull request merged**